crates/forge/src/lib.rs

```diff
@@ -578,72 +494,22 @@ fn run_with_fuzzing(
                 .count(),
         )?;
 
+        if let Some(TestCaseSummary::Passed { .. }) = final_result {
+            if runs != fuzzer_runs {
+                final_result = results
+                    .iter()
+                    .cloned()
+                    .find(|item| matches!(item, TestCaseSummary::Skipped {}));
+            };
+        };
```

Comment:
We have the result extraction logic defined in couple different places, I think this can be simplified. 

1. If there is some `Failed` test, return `Failed`
2. Else if there are some skipped tests, return `Skipped`
3. Else if there are some passed tests, return `Passed`
4. Else if there's no result, panic

Is this flow missing any edge cases?

Reply:
3. if there are only passed tests return `Passed`

Reply:
Why on the type level it is posssible to final result be none 🤔

Reply:
```
        let mut final_result = None;

        while let Some(task) = tasks.next().await {
            let result = task??;

            results.push(result.clone());
            final_result = Some(result.clone());

            if let TestCaseSummary::Failed { .. } = result {
                rec.close();
                break;
            }
        }
```
TBH, I have no idea, how to do it without None. If I remove the option, the compiler claims that None is possible.I think the reason is `tasks` could be empty

Reply:
I have no idea, how to write it better each solution has some disadvantages